src/java/main/org/apache/zookeeper/server/quorum/QuorumPeer.java

```diff
@@ -2069,4 +2073,9 @@ public QuorumCnxManager createCnxnManager() {
                 this.quorumCnxnThreadsSize,
                 this.isQuorumSaslAuthEnabled());
     }
+
+    boolean isLeader(long id) {
+        Vote vote = getCurrentVote();
+        return vote != null && id == vote.getId();
```

Comment:
You could use `state` property to check if the peer is the leader:
```java
return state == ServerState.LEADING;
```
What do you think?

Reply:
I think it would be good to add unit test to verify this logic too.

Reply:
@anmolnar
Do I have to check in the current view?
Is this about temporary leader election phase? I don't know the code very well

 can you please give an hint about this new test?

Do I have to simply test the check I am adding? Like a basic unit test with mockito?

Reply:
I'd second the request to sidestep the election logic for this method if possible as it makes the method harder to reason about. 

It would be nice if the method implied the server was currently performing active leader actions and the current way also covers a server preparing for or nominating itself to be leader (or at least it looks like it). For example, with this code on a 5 server ensemble with one server dead, if one server switches its vote midway through an election epoch then you could be displaying two "leaders" by vote but this is not the same as a two leader splitbrain scenario.

Reply:
@enixon sorry,  to me it is not clear if you are saying that current method is okay or that I should change it according to @anmolnar idea.

For LocalPeerBean it may have sense to also check local election state, but for RemotePeerBean I can't see the meaning.

Alternatively we could return 'false' in case that local peer is performing leader election.

I think we should keep the logic simple

Reply:
@eolivelli I missed the purpose of this method and had assumed that it was strictly concerned with determining whether the local peer was a leader or not. In order to capture the your purpose with the method, I would have to write something much more convoluted:

`    boolean isLeader(long id) {
        long leaderId = -1;
        ZooKeeperServer server = getActiveServer();
        if (server instanceof LeaderZooKeeperServer) {
            leaderId = myid;
        } else if (server instanceof FollowerZooKeeperServer) {
            leaderId = ((FollowerZooKeeperServer)server).getFollower().findLeader().id;
        } else if (server instanceof ObserverZooKeeperServer) {
            leaderId = ((ObserverZooKeeperServer)server).getObserver().findLeader().id;
        }
        return leaderId == id;
    }`

Your solution is much cleaner.